### PR TITLE
fix blob GC test

### PR DESCRIPTION
**Pull request by wes-carlson**
This was caused by #10621, which added support for blob handles using local IDs while offline. Due to a race condition, the container would not always be connected and uploadBlob() would return a local ID handle instead of one using the storage ID as the test expected, which would cause it to fail. I hit similar issues in other tests in the PR, but this seems to only happen against ODSP, not when run locally.

https://github.com/microsoft/FluidFramework/pull/10621/files#diff-aa6ab18dfe191259e467ea83ba348405ff244fbc66f8d5dfe2fc0d7ec62dcbf7

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/blobs.spec.ts [hunk 1]
```diff
@@ -627,7 +635,11 @@ describeNoCompat("Garbage collection of blobs", (getTestObjectProvider) => {
             // Attach the container after the blob is uploaded.
             await container.attach(provider.driver.createCreateNewRequest(provider.documentId));
 
-            // Upload the same blob. This will get de-duped and we will get back a handle with the stoageId instead of
+            // Make sure we are connected or we may get a local ID handle
+            defaultDataStore._root.set("make container connect in", "write mode");
+            await ensureContainerConnectedWriteMode(container as Container);
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/blobs.spec.ts:640]: Just waiting for connected should be enough here right. You don't need to wait for write mode explicitly.

**Comment by agarwal-navin** [comment]
> This was caused by #10621, which added support for blob handles using local IDs while offline. Due to a race condition, the container would not always be connected and uploadBlob() would return a local ID handle instead of one using the storage ID as the test expected, which would cause it to fail. I hit similar issues in other tests in the PR, but this seems to only happen against ODSP, not when run locally.
> 
> https://github.com/microsoft/FluidFramework/pull/10621/files#diff-aa6ab18dfe191259e467ea83ba348405ff244fbc66f8d5dfe2fc0d7ec62dcbf7

Should the test be updated to handle the scenario added by #10621? Do we have tests that cover them?

File: packages/test/test-end-to-end-tests/src/test/blobs.spec.ts [hunk 2]
```diff
@@ -537,6 +545,9 @@ describeNoCompat("Garbage collection of blobs", (getTestObjectProvider) => {
             // Attach the container after the blob is uploaded.
             await container.attach(provider.driver.createCreateNewRequest(provider.documentId));
 
+            // GC requires at least one op to have been processed; it will not work if we remain in read mode
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/blobs.spec.ts:548]:
```suggestion
            // GC requires at least ope op to have been processed. It needs a server timestamp and uses the timestamp of the op.
```

**Review by agarwal-navin — APPROVED** [review]

**Comment by wes-carlson** [comment]
> Should the test be updated to handle the scenario added by https://github.com/microsoft/FluidFramework/pull/10621? Do we have tests that cover them?

There are no GC tests, no. It might be good to add some. The functionality is similar to behavior in detached containers.

**Pull request merged**